browser/brave_rewards/rewards_tab_helper.cc

```diff
@@ -192,13 +210,45 @@ void RewardsTabHelper::MaybeSavePublisherInfo() {
   }
 
   // The Rewards system currently assumes that the |publisher_info| table is
-  // populated by calling `GetPublisherActivityFromUrl` as the user nativates
+  // populated by calling `GetPublisherActivityFromUrl` as the user navigates
   // the web. Previously, this was accomplished within the background script of
   // the Rewards extension.
   rewards_service_->GetPublisherActivityFromUrl(
       tab_id_.id(), GetWebContents().GetLastCommittedURL().spec(), "", "");
 }
 
+void RewardsTabHelper::OnCreatorDetected(
+    GURL url,
+    std::optional<CreatorDetectionScriptInjector::Result> result) {
+  if (!rewards_service_ || !result ||
```

Comment:
I think this should not be possible, right? It would never trigger the script in the first place if rewards_service_ is null so this should be a `CHECK(rewards_service_)`

Reply:
I guess to be extra cautious maybe a DCHECK instead of CHECK for now and we can convert to a CHECK later

Reply:
Converted to a `CHECK`.